server/engine/src/main/java/org/eclipse/lsp/cobol/core/model/VariableUsageUtils.java

```diff
@@ -136,27 +137,27 @@ private boolean checkFirstInSecond(VariableNode node1, VariableNode node2) {
   }
 
   /**
-   * Retrieves {@link VariableUsageNode} based on the passed parent node and variable locality
+   * Retrieves {@link UsageNode} based on the passed parent node and variable locality
    * @param parentNode parent node under which usage needs to be determined
    * @param identifier {@link VariableNameAndLocality} variable locality whose usage needs to be determined.
-   * @return List of {@link VariableUsageNode} for the passed identifier
+   * @return List of {@link UsageNode} for the passed identifier
    */
-  public List<VariableUsageNode> getVariableUsageNode(
+  public List<DefinedAndUsedStructure> getVariableUsageNode(
           Node parentNode, VariableNameAndLocality identifier) {
     return parentNode.getDepthFirstStream()
-            .filter(VariableUsageNode.class::isInstance)
-            .map(VariableUsageNode.class::cast)
+            .filter(UsageNode.class::isInstance)
```

Comment:
looks suspicious? what would happen to the dialect nodes which still uses VariableUsageNode ?